src/runtime_src/core/common/device.h

```diff
@@ -221,6 +221,26 @@ class device : public ishim
     return reinterpret_cast<SectionType>(get_axlf_section_or_error(section, xclbin_id));
   }
 
+  /**
+   * get_axlf_section() - Get section from the file passed in
+   *
+   * filename: file containing the axlf section
+   *
+   * Return: pair of section data and size in bytes
+   */
+  std::pair<const char*, size_t>
+  get_axlf_section(const std::string& filename, axlf_section_kind section) const;
+
+  /**
+   * get_uuids() - Get UUIDs from the axlf section
+   *
+   * dtbuf: axlf section to be parsed
+   *
+   * Return: list of UUIDs
+   */
+  std::vector<std::string>
+  get_uuids(const void *dtbuf) const;
+
```

Comment:
Should not be part of xrt::core_device unless it using the device object some how, I couldn't find that.